modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -2112,7 +2112,7 @@ class TaskProcessor {
 
             if( task.type == ScriptType.SCRIPTLET ) {
                 if( task.exitStatus == Integer.MAX_VALUE )
-                    throw new ProcessFailedException("Process `${task.name}` terminated for an unknown reason -- Likely it has been terminated by the external system")
+                    throw new NodeTerminationException("Process `${task.name}` terminated for an unknown reason -- Likely it has been terminated by the external system")
```

Comment:
Not sure about this tho. The reason of this error is not well defined. Think it should `ProcessFailedException` and retried is there's a NF error condition

Reply:
I think this happens if node with the task is rebooted. It reports max exit status, if I remember corectly.

Reply:
The question is whether this exit code is used for other error cases? As I recall, an exit code of `-1`  (aka `255`) is used for many types of errors. It may also be used differently by different executors. Keep in mind this code segment is not specific to k8s.

Reply:
perhaps we solve this issue impelementing jobs instead of pods.

Reply:
Sounds good. @phue Please revert your changes in `TaskProcessor.groovy`. We will merge the k8s specific change now, and we aim to address the other error in a different PR. See #2424 .